### DISCUSSION TITLE
`watch` doesn't unfold fields to object when watching all fields.

### DISCUSSION THREAD

**Discussion by talhabalaj** [react-hook-form/react-hook-form#1795] [Q&A] [answered]
```javascript
const watchAllFields = watch();
// Outputs
// { "images[0].url": "...." } 
// while
const watchImages = watch("images");
// { "images": [ {url: "..."  ] }
```

How can I force `watch()` to show nested objects when watching all the fields?

**Comment by bluebill1049** [comment, marked as answer]
`watch({ nest: true })` we are fixing this in the next major release.

**Reply by talhabalaj** [reply]
is `watch({ nest: true })` available right now?

**Reply by bluebill1049** [reply]
it's available now: https://react-hook-form.com/api#watch but will be removed and return nested value by default in v6, you can try v.6.0.0-rc.2

**Reply by talhabalaj** [reply]
Great, thank you ❤️

**Reply by bluebill1049** [reply]
here is the sneak peek for v6 (if you interested) https://react-hook-form-website-git-v6.bluebill1049.now.sh/migrate-v5-to-v6

**Reply by talhabalaj** [reply]
AWWEESOOOOM!!!
